src/http_server.rs

```diff
@@ -130,16 +132,27 @@ async fn get_network_state(Extension(client): Extension<Arc<NetworkClient>>) ->
     axum::Json(client.network_state())
 }
 
+async fn get_config(Extension(config): Extension<Arc<Config>>) -> impl IntoResponse {
+    axum::Json((*config).clone())
+}
+
+async fn get_dataset_state(
+    Path((dataset)): Path<(String)>,
+    Extension(client): Extension<Arc<NetworkClient>>,
+    Extension(config): Extension<Arc<Config>>,
+)  -> impl IntoResponse {
+    let Some(dataset_id) = config.dataset_id(&dataset) else {
+        return RequestError::NotFound(format!("Unknown dataset: {dataset}")).into_response();
+    };
+
+    axum::Json(client.dataset_state(dataset_id)).into_response()
+}
+
 async fn get_metrics(Extension(registry): Extension<Arc<Registry>>) -> impl IntoResponse {
     lazy_static::lazy_static! {
         static ref HEADERS: HeaderMap = {
             let mut headers = HeaderMap::new();
-            headers.insert(
-                header::CONTENT_TYPE,
-                "application/openmetrics-text; version=1.0.0; charset=utf-8"
-                    .parse()
-                    .unwrap(),
-            );
+            headers.insert(header::CONTENT_TYPE, "text/plain; charset=utf-8".parse().unwrap());
```

Comment:
It will break prometheus collection with a very unintuitive error. And that's the default content type, you don't even have to specify it.

```
prometheus {"caller":"scrape.go:1399","component":"scrape manager","err":"invalid metric type \"info\"","level":"debug","msg":"Append failed","scrape_pool":"scrapeconfig/prometheus-external/http","target":"http://95.216.245.20:9092/metrics","ts":"2024-04-24T13:02:21.989Z"}
```
https://github.com/prometheus/client_rust/issues/111